src/server/config/complete.js

```diff
@@ -1,8 +1,18 @@
 import { difference, keys } from 'lodash';
 import { transformDeprecations } from './transform_deprecations';
+import flattenWith from './flatten_with';
+import { formatListAsProse } from '../../utils';
 
-const getUnusedSettings = (settings, configValues) => {
-  return difference(keys(transformDeprecations(settings)), keys(configValues));
+const getUnusedConfigKeys = (settings, configValues) => {
+  const inputKeys = keys(flattenWith('.', transformDeprecations(settings)));
+  if (inputKeys.includes('env')) {
```

Comment:
__question:__ it feels that this new `env` related case deserves a dedicated unit test. Could you please add  something to cover it?